File: ransom-note/ransom-note.js
let canConstruct = function(ransomNote, magazine) {
    let ransomMap = {}
    for (let i = 0; i < ransomNote.length; i++) {
        let current = ransomNote[i]
        if (current in ransomMap) {
            ransomMap[current]++
        } else {
            ransomMap[current] = 1
        }
    }

    for (let i = 0; i < magazine.length; i++) {
        let current = magazine[i]
        if (current in ransomMap) {
            ransomMap[current]--
            if (ransomMap[current] < 1) {
                delete ransomMap[current]
            }

            if (!Object.keys(ransomMap).length) {
                return true
            }
        }
    }

    return !Object.keys(ransomMap).length
};

File: ransom-note/ransom-note.py
def canConstruct(ransomNote, magazine):
    ransomMap = {}
    for character in ransomNote:
        if character in ransomMap:
            ransomMap[character] += 1
        else:
            ransomMap[character] = 1

    for character in magazine:
        if character in ransomMap:
            ransomMap[character] -= 1
            if ransomMap[character] < 1:
                del ransomMap[character]

            if not bool(ransomMap):
                return True

    return not bool(ransomMap)